shamil_latypov/main.cpp

```diff
@@ -364,28 +360,24 @@ void generate_yellow_edges(Graph& graph) {
                   (double)(depth - graph.get_depth_map().begin())) {
         // Выбираем вершину уровнем глубже, которая не является
         // потомком нашей вершины
-        std::uniform_int_distribution<> rand_vertex(
-            0, vert_ids_depth_deeper.size() - 1);
 
-        graph.add_edge(vertex_id, vert_ids_depth_deeper[rand_vertex(gen)]);
+        graph.add_edge(vertex_id,
+                       vert_ids_depth_deeper[random_number() %
+                                             vert_ids_depth_deeper.size()]);
```

Comment:
Странно, опять мой комментарий пропал... Что за ерунда, не понимаю...
Ты сейчас написал `C-Style` рандомную генерацию, до этого было лучше.
Писал я тебе следующее:
Чтобы убрать из функции низкоуровневые детали реализации и упростить понимание того. что она делает. А опять же убрать дублирование между последжними 2умя функциями, лучше будет эту логику вынести в отдельную функцию:
```suggestion
        graph.add_edge(vertex_id, get_random_vertex_id(vert_ids_depth_deeper));
```
Соответственно в `generate_red_edges` так же должно быть